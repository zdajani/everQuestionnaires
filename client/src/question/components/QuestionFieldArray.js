import React from 'react';
import { Field } from 'redux-form';
import renderField from '../../commonComponents/Field';
import './styles/QuestionFieldArray.css'

const QuestionFieldArray = ({ fields, meta: {error, submitFailed} }) => (
  <div>
    <div className="form-group">
    <button type="button" className="btn btn-outline-info btn-lg" onClick={() => fields.push({})}>Add Question</button>
      {submitFailed &&
      error &&
      <span className="p-3 mb-2 bg-white text-danger" >
          {error}
      </span>}
    </div>
      
    {fields.map((question, index) => (
      <div key={index}>
        <div className="form-group add-question-title">      
          <h4> Question #{index + 1}  
            <span className="remove-button">
              <button type="button" 
                title="Remove Question" 
                onClick={() => fields.remove(index)}
                className="btn btn-outline-danger">
                  Remove
              </button>
            </span> 
          </h4>
        </div>  
        
        <div className="form-group">          
          <Field
            name={`${question}.name`}
            type="text"
            component={renderField}
            label="Name"
          />
        </div>
        <div className="form-group">      
          <Field
            name={`${question}.label`}
            type="text"
            component={renderField}
            label="Label"
            isTextArea={true}
          />
          </div>
        </div>  
      ))}
  </div>
);

export default QuestionFieldArray;