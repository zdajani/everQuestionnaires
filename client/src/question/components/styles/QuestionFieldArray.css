.remove-button {
  padding-left: 1rem;
}

button.btn.btn-link.warning {
  color: red;
  font-size: 1.5
}

.add-question-title {
  padding-top: 1.5rem;
}