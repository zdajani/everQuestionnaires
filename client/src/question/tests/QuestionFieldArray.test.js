import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import QuestionFieldArray from '../../question/components/QuestionFieldArray';

describe('QuestionnaireForm', () => {
  it('renders self without crashing', () => { 
    const meta = {error: {}, submitFailed: false}
    const component = shallow(<QuestionFieldArray fields={[]} meta={meta}  />)
    expect(toJson(component)).toMatchSnapshot();
  });
})