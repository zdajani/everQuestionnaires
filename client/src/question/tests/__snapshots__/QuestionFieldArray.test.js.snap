// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionnaireForm renders self without crashing 1`] = `
<div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-outline-info btn-lg"
      onClick={[Function]}
      type="button"
    >
      Add Question
    </button>
  </div>
</div>
`;
